main.tf

```diff
@@ -175,6 +177,27 @@ resource "aws_apigatewayv2_integration" "this" {
   }
 }
 
+# Authorizers
+resource "aws_apigatewayv2_authorizer" "this" {
+  for_each = var.create && var.create_routes_and_integrations ? var.authorizers : {}
+
+  api_id    = aws_apigatewayv2_api.this[0].id
+
+  authorizer_type  = lookup(each.value, "authorizer_type", null)
+  identity_sources = [lookup(each.value, "identity_sources", null)]
+  name             = lookup(each.value, "name", null)
+  authorizer_uri   = lookup(each.value, "authorizer_uri", null)
+  authorizer_payload_format_version = lookup(each.value, "authorizer_payload_format_version", null)
+
+  dynamic "jwt_configuration" {
+    for_each = lookup(each.value, "audience", null) == null ? tolist([]) : tolist([lookup(each.value, "audience", null)])
+    content {
+      audience = [lookup(each.value, "audience", null)]
```

Comment:
```suggestion
      audience = jwt_configuration.value
```

Reply:
TBH - Both sub-elements (issuer and audience are both optional - the for_each should be based on the either of them being present, but not necessarily both...

Reply:
I made this change assuming that audience and issuer would both be present. But it's possible to supply just issuer.

Reply:
Then let's change `for_each` on line 193 to handle that at least one value (audience or issuer) should be present.